src/tmpfiles/tmpfiles.c

```diff
@@ -862,15 +862,34 @@ static int path_open_parent_safe(const char *path) {
         return fd;
 }
 
+static int path_open_safe(const char *path) {
+        int fd;
+
+        /* path_open_safe() returns a file descriptor opened with O_PATH after
+         * verifying that the path doesn't contain unsafe transitions, except
+         * for its final component as the function does not follow symlink. */
+
+        assert(path);
+
+        if (path_equal(path, "/") || !path_is_normalized(path))
+                return -EINVAL;
```

Comment:
is there any reason not to allow the root dir here?

this means people can't use tmpfiles to let's say add an xattr on the root dir. Why prohibit that though?

Reply:
Indeed, I'll remove it.